core/iwasm/libraries/lib-socket/src/wasi/wasi_socket_ext.c

```diff
@@ -143,7 +143,7 @@ accept(int sockfd, struct sockaddr *addr, socklen_t *addrlen)
     error = __wasi_sock_accept(sockfd, 0, &new_sockfd);
     HANDLE_ERROR(error)
 
-    error = getpeername(new_sockfd, addr, addrlen);
```

Comment:
Actually, this should be handled in a different way. `getpeername` already sets `errno` and returns `-1` on error, so I passing it later to `HANDLE_ERROR` might override the correct `errno` code. I think the correct handling should be like:
```
if (getpeername(new_sockfd, addr, addrlen) == -1) {
  return -1;
}
```

Reply:
Yes, seems it is better to change the code like you said:
```
RETURN VALUE
       On success, zero is returned.  On error, -1 is returned, and errno is set appropriately.
```

Reply:
Indeed, nicely catch. The change has been applied to this PR.